src/components/Code/Code.tsx

```diff
@@ -0,0 +1,125 @@
+import { CopyIcon, CheckIcon } from "@chakra-ui/icons";
+import { Box, Flex, IconButton, useClipboard } from "@chakra-ui/react";
+import Highlight, { defaultProps, Language } from "prism-react-renderer";
+import nightOwlLight from "prism-react-renderer/themes/nightOwlLight";
+import { ComponentPropsWithoutRef, FunctionComponent } from "react";
+
+interface CodeProps {
+  code: string;
+  language: Language;
+}
+
+// Unfortunately not exported by prism-react-renderer
+type RenderProps = Parameters<
+  ComponentPropsWithoutRef<typeof Highlight>["children"]
+>[0];
+
+const btnThemes = {
+  singleLine: {
+    colorScheme: "blue",
+    color: "blue.500",
+    ml: 4,
+  },
+  multiLine: {
+    color: "white",
+    sx: {
+      ":hover": {
+        bg: "blue.400",
+      },
+    },
+  },
+};
+
+const CopyButton: FunctionComponent<{
```

Comment:
I'll probably generalize this component this component in a later commit, this PR is in draft since it still needs a bit of cleanup as some stuff is still slightly rushed